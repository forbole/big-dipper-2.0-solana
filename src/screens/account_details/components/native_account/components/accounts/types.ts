export type AddressesType = string[];

export type AccountsType = {
  stake: AddressesType;
  nonce: AddressesType;
  vote: AddressesType;
  token: AddressesType;
}
