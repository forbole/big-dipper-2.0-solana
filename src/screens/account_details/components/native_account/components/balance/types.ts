export type BalanceType = {
  native: TokenUnit;
  stake: TokenUnit;
  nonce: TokenUnit;
  vote: TokenUnit;
  total: TokenUnit;
}
