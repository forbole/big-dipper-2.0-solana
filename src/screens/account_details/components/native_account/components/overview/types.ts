export type OverviewType = {
  address: string;
}
