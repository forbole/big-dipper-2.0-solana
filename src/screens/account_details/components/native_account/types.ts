export type OverviewType = {
  address: string;
}

export type AccountType = {
  address: string;
}

export type AccountsType = {
  data: {
    [value:number]: AccountType[];
  };
  count: number;
  loading: boolean;
}

export type NativeAccountState = {
  loading: boolean;
  exists: boolean;
  overview: OverviewType;
}
