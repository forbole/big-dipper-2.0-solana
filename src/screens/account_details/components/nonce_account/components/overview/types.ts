export type OverviewType = {
  address: string;
  balance: number;
  authority: string;
  blockhash: string;
  fee: number;
}
