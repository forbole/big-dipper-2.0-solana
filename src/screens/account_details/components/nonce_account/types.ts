export type OverviewType = {
  address: string;
  authority: string;
  balance: TokenUnit;
  blockhash: string;
  fee: number;
}

export type NonceAccountState = {
  loading: boolean;
  overview: OverviewType;
}
