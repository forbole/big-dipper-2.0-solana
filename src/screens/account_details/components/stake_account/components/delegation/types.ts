export type DelegationType = {
  activated: TokenUnit;
  activating: TokenUnit;
  total: TokenUnit;
}
