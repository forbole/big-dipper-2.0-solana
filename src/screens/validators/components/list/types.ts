// ryuash
// export type ValidatorType = {
//   validator: string;
//   stake: number;
//   stakePercent: number;
//   fee: number;
//   lastVote: number;
//   skipRate: number;
//   skipTotal: number;
//   skipPercent: number;
//   condition: number;
//   status: number;
//   delegators: number;
// }

export type ValidatorType = {
  validator: string;
  commission: number;
  stake: string;
  stakePercent: string;
  lastVote: number;
  status: boolean;
}

export type ValidatorsState = {
  loading: boolean;
  exists: boolean;
  tab: number;
  sortKey: string;
  sortDirection: 'asc' | 'desc';
  votingPowerOverall: number;
  items: ValidatorType[];
}

export type ItemType = Override<ValidatorType, { validator: AvatarName }>;
