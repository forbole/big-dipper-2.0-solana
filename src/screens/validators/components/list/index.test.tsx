/* eslint-disable */
import React from 'react';
import { RecoilRoot } from 'recoil';
import renderer from 'react-test-renderer';
import {
  MockTheme, wait,
} from '@tests/utils';
import { createMockClient } from 'mock-apollo-client';
import { ApolloProvider } from '@apollo/client';
import { ValidatorsDocument } from '@graphql/types';
import List from '.';

// ==================================
// mocks
// ==================================
jest.mock('./components', () => ({
  Mobile: (props) => <div id="Mobile" {...props} />,
  Desktop: (props) => <div id="Desktop" {...props} />,
  Tabs: (props) => <div id="Tabs" {...props} />,
}));

jest.mock('@components', () => ({
  Box: (props) => <div id="Box" {...props} />,
  NoData: (props) => <div id="NoData" {...props} />,
  LoadAndExist: (props) => <div id="LoadAndExist" {...props} />,
  TabPanel: (props) => <div id="TabPanel" {...props} />,
}));

const mockValidatorsDocument = jest.fn().mockResolvedValue({
  data: {
    "validator": [
      {
        "address": "BxTUwfMiokzimVDLDupGfVPmWXfLSGVpkGr9TUmetn6b",
        "commission": 10,
        "validatorStatus": {
          "active": true,
          "activatedStake": 96138469659982,
          "lastVote": 115025839
        }
      }
    ]
  },
});

// ==================================
// unit tests
// ==================================
describe('screen: Validators/List', () => {
  it('matches snapshot', async () => {
    const mockClient = createMockClient();
    mockClient.setRequestHandler(
      ValidatorsDocument,
      mockValidatorsDocument,
    );

    let component;

    renderer.act(() => {
      component = renderer.create(
        <RecoilRoot>
          <ApolloProvider client={mockClient}>
            <MockTheme>
              <List />
            </MockTheme>
          </ApolloProvider>
        </RecoilRoot>,
      );
    });
    await wait();

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  afterEach(() => {
    jest.clearAllMocks();
  });
});
