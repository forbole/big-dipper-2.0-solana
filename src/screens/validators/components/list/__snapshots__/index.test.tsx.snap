// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`screen: Validators/List matches snapshot 1`] = `
<div
  exists={true}
  id="LoadAndExist"
  loading={false}
>
  <div
    className=""
    id="Box"
  >
    <div
      handleSearch={[Function]}
      handleTabChange={[Function]}
      id="Tabs"
      tab={0}
      validatorCount={
        Array [
          "1",
          "0",
          "0",
        ]
      }
    />
    <div
      id="TabPanel"
      index={0}
      value={0}
    >
      <div
        className="makeStyles-list"
      />
    </div>
    <div
      id="TabPanel"
      index={1}
      value={0}
    >
      <div
        className="makeStyles-list"
      />
    </div>
    <div
      id="TabPanel"
      index={2}
      value={0}
    >
      <div
        className="makeStyles-list"
      />
    </div>
  </div>
</div>
`;
